tools/ThermoRawFileParser/thermo_converter.xml

```diff
@@ -1,145 +1,202 @@
+ln -s -f '$input_file' '$input_name' &&
 
 ThermoRawFileParser.sh
-    -d=./raws_folder
-    -o=./output_folder
-    -f=$output_format
+    --input='$input_name'
+    --output_file='$output'
+    -f=$format_cond.output_format
     #if $output_metadata_selector != "off":
-        --metadata="${output_metadata_selector}"
+        --metadata="$output_metadata_selector"
+        --metadata_output_file='$output_metadata'
+    #end if
+    #if $format_cond.output_format == "0"
+        $format_cond.mgfPrecursor
+    #end if
+    #if $format_cond.output_format in ['1', '2']
+        $format_cond.zlib_boolean
+        $format_cond.peakpicking_boolean
     #end if
-    $zlib_boolean
-    $peakpicking_boolean
     $ignore_instrument_errors_boolean
-
+    $allDetectors
+    $includeExceptionData
+    #if $msLevel != ''
+        --msLevel="$msLevel"
+    #end if
 ]]>
     </command>
     <inputs>
-        <param name="input_files" type="data" format="thermo.raw" label="Thermo RAW file" help="" multiple="true"
-          optional="False" />
-
-        <param name="output_format" type="select" label="Output format">
-            <option value="0">mgf</option>
-            <option value="1" selected="True">mzml</option>
-            <option value="2">Indexed mzml</option>
-        </param>
-
-        <param name="zlib_boolean" type="boolean" truevalue="" falsevalue="-z" checked="true"
-            label="Use zlib compression for the m/z ratios and intensities" help="" />
-
-        <param name="peakpicking_boolean" type="boolean" truevalue="" falsevalue="-p" checked="true"
-            label="Use the peak picking provided by the native thermo library" help="" />
-
-        <param name="ignore_instrument_errors_boolean" type="boolean" truevalue="-e" falsevalue="" checked="true"
-            label="Ignore missing instrument properties" help="If false, it stops the conversion if instrument properties are missing" />
-
-        <param name="output_metadata_selector" type="select" label="Output metadata" >
+        <param argument="--input" name="input_file" type="data" format="thermo.raw" label="Thermo RAW file" help="" optional="False"/>
+        <conditional name="format_cond">
+            <param argument="--format" name="output_format" type="select" label="Output format">
+                <option value="0">mgf</option>
+                <option value="1" selected="True">mzml</option>
+                <option value="2">Indexed mzml</option>
+                <!-- <option value="3">Parquet</option> -->
+            </param>
+            <when value="0">
+                <param argument="--mgfPrecursor" type="boolean" truevalue="--mgfPrecursor" falsevalue="" label="Include precursor scan number in MGF file" help=""/>
+            </when>
+            <when value="1">
+                <param argument="--noZlibCompression" name="zlib_boolean" type="boolean" truevalue="" falsevalue="-z" checked="true"
```

Comment:
I'm wondering what this implies? Is there a use-case for our users to disable it?

Reply:
As far as I know different compression schemes can be used for numerical values. Maybe one or the other downstream tool is incompatible with the default ..